File: 151001/Aleksandrov/RESTLab1/src/main/java/by/bsuir/entities/Issue.java
package by.bsuir.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Issue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long editorId;
    private String title;
    private String content;
    private Timestamp created;
    private Timestamp modified;
    @ManyToOne
    @JoinColumn(name = "editorId", insertable = false, updatable = false)
    private Editor editor;
    @ManyToMany
    @JoinTable(
            name = "issuelabel",
            joinColumns = @JoinColumn(name = "issueId"),
            inverseJoinColumns = @JoinColumn(name = "labelId")
    )
    private Set<Label> labels;
}

File: 151004/Bordilovsky/DC_REST/Entities/Issue.cs
namespace DC_REST.Entities
{
	public class Issue
	{
		public int Id { get; set; }

		public int UserId { get; set; }
		public string Title { get; set; }
		public string Content { get; set; }
		public DateTime Created { get; set; }
		public DateTime Modified { get; set; }

		public User User { get; set; }
		//public List<Note>? Notes { get; set; } = new();
		//public List<Label> Labels { get; set; } = new();
		public List<Issue_Label> Issue_Labels { get; set; } = new();
	}
}
